frontend/apps/reader/modules/readerview.lua

```diff
@@ -487,6 +493,57 @@ function ReaderView:drawTempHighlight(bb, x, y)
 end
 
 function ReaderView:drawSavedHighlight(bb, x, y)
+    -- set/free note mark sign
+    if self.highlight.note_mark == "sidemark" then
+        if not self.note_mark_sign then
+            self.note_mark_sign = TextWidget:new{
+                text = "\u{F040}", -- pencil
+                face = Font:getFace("smallinfofont", 14),
+                padding = 0,
+            }
+        end
+    else
+        if self.note_mark_sign then
+            self.note_mark_sign:free()
+            self.note_mark_sign = nil
+        end
+    end
+    -- calculate position x of the note side line/mark
+    if self.highlight.note_mark == "sideline" or self.highlight.note_mark == "sidemark" then
+        local screen_w = Screen:getWidth()
+        local sign_w = self.highlight.note_mark == "sideline" and 3 or self.note_mark_sign:getWidth()
+        local sign_gap
+        if self.ui.paging then
+            sign_gap = Screen:scaleBySize(10) -- to screen edge (fixed position)
```

Comment:
I guess this `10` may have the marker overlap with the text. No way to take the PDF margins setting (from the bottom menu) into account ?

Reply:
pdf margins are not applicable in some zoom modes.

![10](https://user-images.githubusercontent.com/62179190/182172355-0ea771c2-afe1-493d-a8eb-724c8e993498.png)

So the line/mark placement is done fixed and as close to the screen edge as possible (maybe change 10 to 5).
Should the position be flexible? (it will require to determine the pdf view/zoom mode, margins etc)
This will not guarantee no overlapping.

Reply:
Feels complicated indeed :)
May be push it nearer the screen edge:
If the line thickness is 3, add a gap of 3, just enough to distinguish it from the edge.
If the pencil is N pixels wide, keep a gap of N/4 or N/6 or N/8 - or just 3 - to keep a bit of blank.